ext/RMagick/rmimage.c

```diff
@@ -9787,29 +10776,45 @@ Image_pixel_color(int argc, VALUE *argv, VALUE self)
         return Pixel_from_PixelColor(&image->background_color);
     }
 
+    exception = AcquireExceptionInfo();
+
     if (image->storage_class == PseudoClass)
     {
+#if defined(IMAGEMAGICK_7)
+        okay = SetImageStorageClass(image, DirectClass, exception);
+#else
         okay = SetImageStorageClass(image, DirectClass);
         rm_check_image_exception(image, RetainOnError);
+#endif
         if (!okay)
         {
             rb_raise(Class_ImageMagickError, "SetImageStorageClass failed. Can't set pixel color.");
```

Comment:
Before this line, it needs to destroy `exception` to avoid memory leak.

Reply:
`exception` will only be created inside an `#if defined(IMAGEMAGICK_7)` block?

Reply:
Aaah here was the comment :) Edited your message mockdeep and this is indeed a memory leak thanks for finding this @Watson1978.